src/Raven.Server/Documents/Handlers/Processors/Queries/AbstractOperationQueriesHandlerProcessor.cs

```diff
@@ -60,14 +61,26 @@ public override async ValueTask ExecuteAsync()
 
     protected QueryOperationOptions GetQueryOperationOptions()
     {
-        return new QueryOperationOptions
+        var options = new QueryOperationOptions
         {
             AllowStale = RequestHandler.GetBoolValueQueryString("allowStale", required: false) ?? false,
             MaxOpsPerSecond = RequestHandler.GetIntValueQueryString("maxOpsPerSec", required: false),
             StaleTimeout = RequestHandler.GetTimeSpanQueryString("staleTimeout", required: false),
             RetrieveDetails = RequestHandler.GetBoolValueQueryString("details", required: false) ?? false,
-            IgnoreMaxStepsForScript = RequestHandler.GetBoolValueQueryString("ignoreMaxStepsForScript", required: false) ?? false
+            IgnoreMaxStepsForScript = RequestHandler.GetBoolValueQueryString("ignoreMaxStepsForScript", required: false) ?? false,
         };
+        var waitForIndexes = RequestHandler.GetBoolValueQueryString("waitForIndexes", required: false) ?? false;
```

Comment:
I do not think this one is needed, check how this is handled in the batch handler